Updated Docs

burrrata opened the pull request:
Added more detailed explanations to the user guide.
Moved the "adding to a dao" section after the "user guide" section so that it's more in flow with the doc (install, learn/try, install).
Added a few gifs to show people how to use the app (although it looks like @lkngtn already added some?).

fabriziovigevani commented on README.md:
```diff
@@ -74,36 +74,65 @@ ETH 0x0000000000000000000000000000000000000000 Balance: 2
+The tokens you can add to the Redemptions app are dependent on the blockchain you're on. If you're on the Ethereum mainnet you can add any erc-20 token contract. If you're on a local testnet you can add any token contracts that have been deployed locally. 
 
-To deploy to an organization you can use the [Aragon CLI](https://hack.aragon.org/docs/cli-intro.html).
+Who/what is allowed to control the Redemptions app is determined by the DAO's permissions. You can check the permissions for your DAO in the `Permissions` tab of the Aragon client. Every app currently installed should be list. Click on an app to view it's permissions. 
```
```suggestion
Who/what is allowed to control the Redemptions app is determined by the DAO's permissions. You can check the permissions for your DAO in the `Permissions` tab of the Aragon client. Every app currently installed should be listed. Click on an app to view it's permissions.

burrrata replied:
Good catch. Thanks :)

fabriziovigevani commented on README.md:
```diff
@@ -74,36 +74,65 @@ ETH 0x0000000000000000000000000000000000000000 Balance: 2
+
+To remove a token from the Redemptions app, open the Redmpetions app in your DAO. Then hover your cursor over the token symbol you would like to remove. You should see the name and amount of the token turn red and display the word "Remove." If you have the Voting app configured to manage token removals (and we recommend this), clicking "Remove" will create a vote to remove the token from Redempti ...
+
+WARNING: if you remove a token from the Redemptions app that has a positive balance, those tokens will be gone. Make sure that you redeem or move all the tokens before taking this action.   
```
This should be removed. 

Token balances listed in Redemptions reflect the vault's balance for that token, not the Redemptions contract balance.

burrrata replied:
I think we should still document what the "Remove" button does, but just change it to be accurate. That's what you meant right?

And then to confirm, the "Remove" button on the Redemptions app just removes the token from the redemptionTokenList and thus the UI of the app, but not from the DAO's Vault? Users can add/remove whatever contracts they want from the Redemptions app UI, but what really matters is what's in the Vault

burrrata replied:
Just pushed a fix to the README that explains that the Remove button in the Redemptions UI just removes it from the UI, but not the Vault.

fabriziovigevani commented on README.md:
```diff
@@ -74,36 +74,65 @@ ETH 0x0000000000000000000000000000000000000000 Balance: 2
-## Using Redemptions
+<p align="center">
+    <img src="https://raw.githubusercontent.com/1Hive/redemptions-app/master/docs/resources/add-token-to-redemptions-app.gif" width="600" />
+</p>
```
If i recall correctly, initially @lkngtn added giffs for adding a removing tokens but in the end decided to remove them as it made the doc too heavy.

I think that with a step by step description like above should do for adding tokens but would keep the current step of clicking on `Add token`, entering token address..

burrrata replied:
Gotcha. So the README is going to be like a quickstart and then more detailed user guides and stuff can live in the `docs` directory?

fabriziovigevani commented on README.md:
````diff
@@ -54,7 +54,7 @@ You will also see the configuration for your local deployment in the terminal. I
 ```
 
-Currently the only thing deployed on your local testnet is an Aragon DAO with the Redemption app. In a new terminal navigate to the `redemptions-app` directory. Then run this truffle script to deploy some token contracts to interact with.
+Currently the only thing deployed on your local testnet is an Aragon DAO with the Redemption app. In a new terminal navigate to the `redemptions-app` directory. Then run this script to deploy some token contracts on your local testnet to interact with.
````
```suggestion
Currently the only thing deployed on your local testnet is an Aragon DAO with the Redemptions app. In a new terminal navigate to the `redemptions-app` directory. Then run this script to deploy some token contracts on your local testnet to interact with.
```

0xGabi commented:
Hey @burrrata I notice that gif on your branch are not displayed correctly on the [Readme file](https://github.com/1Hive/redemptions-app/tree/updated-readme)

burrrata commented:
@0xGabi Yeah the links to the gifs are set to the master branch not the updated-readme branch.They should show up if/when they get merged into the master. That being said, maybe it would be better to keep the README short and sweet and move the gifs to a separate "user guide" doc in the `docs` repo?

lkngtn commented:
I think the README should be strictly for developer and then link to user docs elsewhere. Separating this out into multiple docs I think will also help when we sync the info to the website.

burrrata commented:
Just updated the docs:
- User Guide section moved to `docs/user-guide.md`
- `docs/dev-guide` renamed to `techincal-documentation.md` (to not get confused with the docs folder because docs/docs.md would be weird)
- removed extra junk that was in the docs folder (overview.md and basic-info.md)

TODO: 
- update technical-documentation.md

lkngtn commented:
I like the idea of having the bug bounty information separately, perhaps as part of the contribute section on the website, and then for projects which are eligible include link to that page in their respective docs?

burrrata commented:
Updated the `technical-documentation.md` page. 
- walked through every function in redemptions.sol, describing what it does and what the expected behavior will be

This is not a final draft of the technical documentation.
- does not yet include details on how functions will fail when the expected behavior does not occur
- may or may not have some errors/oversights 
- the formatting is ok, but I'd like to refine the doc to make it more user friendly

burrrata commented:
Removed bug-bounty.md from this PR. Will move to the website

burrrata commented:
Added `overview.md` back into the docs folder and updated it because it's useful to sync to the main website or anywhere that needs a high level overview of all the stuff related to the app (overview paragraph, user guide, docs, bug bounty, etc...)

fabriziovigevani commented on docs/user-guide.md:
```diff
@@ -1,12 +1,50 @@
+### Adding tokens:
+
+The default recommendation is for the Voting app to control adding and removing tokens from the Redemptions app. This way your community can collectively manage the state of Redemptions. Adding tokens to Redemptions via the voting app requires two steps:
+- creating a vote to add the token to the Redeptions app
```
```suggestion
- creating a vote to add the token to the Redemptions app
```

fabriziovigevani commented on docs/user-guide.md:
```diff
@@ -1,12 +1,50 @@
+
+### Removing tokens:
+
+To remove a token from the Redemptions app UI, open the Redmpetions app in your DAO. Then hover your cursor over the token symbol you would like to remove. You should see the name and amount of the token turn red and display the word "Remove." If you have the Voting app configured to manage token removals (and we recommend this), clicking "Remove" will create a vote to remove the token from Redem ...
```
```suggestion
To remove a token from the Redemptions app UI, open the Redemptions app in your DAO. Then hover your cursor over the token symbol you would like to remove. You should see the name and amount of the token turn red and display the word "Remove." If you have the Voting app configured to manage token removals (and we recommend this), clicking "Remove" will create a vote to remove the token from Redemptions app UI. Your community will then need to approve this change for it to pass. 
```

fabriziovigevani commented on docs/user-guide.md:
```diff
@@ -1,12 +1,50 @@
+
+### Redeeming tokens:
+
+To redeem tokens, click on the "Redeem" button, then use the slider to select how many tokens you would like to redeem. When satified with the amount, click redeem to confirm. You will be prompted to sign two messages: one to confirm the transaction and one to send that transaction to the network to be processed. 
```
It should be noted that the amount of tokens to redeem can also be set from the input field.

fabriziovigevani commented on docs/user-guide.md:
```diff
@@ -1,12 +1,50 @@
-### How do you use it?
-- user guide TBD
-- ideally this will have a walkthrough of the apps features with screenshots showing how to interact with the app to do stuff
+The redemptions app allows your organization to create a vault that exchanges your native token for an underlying asset. For example, the 1Hive native token is HONEY, but often times organization members will want DAI or ETH. The redemptions vault allows BEEs to exchange the HONEY they earn by contributing to the hive for the underlying asset of the vault.
```
The app doesn't create a vault. It uses the already created vault of the dao.

burrrata replied:
Good catch. Updated accordingly

fabriziovigevani commented on docs/technical-documentation.md:
```diff
@@ -0,0 +1,119 @@
+
+## Initialization
+
+The Redemptions app is initialized by passing a `Vault _vault`, `TokenManager _tokenManager`, and `address[] _redemptionTokenList`. Users are able to redeem tokens associated to the `_tokenManager` in exchange for a proportional share of each token on the `_redemptionTokenList` held in the `_vault` address.
```
Currently, the app is initialized only with the `Vault` and `TokenManager`. The tokenList is set empty for now until we can use argument arrays from the cli

burrrata replied:
Ok cool. Changed the docs accordingly. If we end up allowing admins to initialize the `_redemptionTokenList` via the CLI let me know

burrrata commented on docs/overview.md:
```diff
@@ -1,30 +1,29 @@
 
-### Maintainer 🚧
-- [1Hive Workers](https://1hive.org/docs/contribute/projects-tasks.html#expectations-of-workers)
+The Redemptions [userguide](docs/user-guide.md) explains how an end user can interact with Redemptions to ad, remove, and redeem tokens.
```
Good call

burrrata commented on docs/technical-documentation.md:
```diff
@@ -0,0 +1,119 @@
+
+## Initialization
+
+The Redemptions app is initialized by passing a `Vault _vault` and `TokenManager _tokenManager` parameters. Users are then able to add tokens to Redemptions via the GUI in the DAO. This passes an Ethereum contract address to the `address[] _redemptionTokenList`array. This then allows users to redeem tokens associated to the `_tokenManager` in exchange for a proportional share of each token on the ...
```
Yeah you're right. That's just a terrible paragraph lol. I'm honestly not sure how that even got in there. My bad

burrrata replied:
Reading through this again and fixing it, and I wonder if I messed up a branch merge or something because it's really like there's just two different paragraphs here. It's just so bad... lol

burrrata commented on docs/technical-documentation.md:
```diff
@@ -0,0 +1,119 @@
+
+## Redeeming Tokens
+
+Anyone account that holds redemption tokens can call the `redeem(uint256 _amount)` function. This will:
```
Good call

burrrata commented:
@lkngtn I think this is good to go if you want to take a look and merge it in

burrrata commented:
`redemptions.sol` was changed since the docs were last updated. Now this branch has conflicts with master. To avoid this (and potentially rewriting the docs) in the future, should we:
- just go ahead and merge changes that only affect the documentation?
- wait to do docs until the codebase is more or less frozen?
- proceed with branches and manually triage and update stuff if necessary?

fabriziovigevani commented:
I see there's a lot of commits i made from separate branches i've worked on. 
Are you merging to your branch every time master is updated or doing a git rebase?

As you are the only one doing changes to the docs, keep your branch as clean as possible and only add your commits so when merging we won't have this kind of conflicts.

In this case we can just resolve the conflict manually as it's a small one.

burrrata commented:
I just created a branch for docs and then kept working on that, but merged a few changes in as the codebase changed so that I could read the code in my text editor. In the future I'll make sure to keep the docs branch completely separated from everything else so that it's easier to merge.

burrrata commented:
Merging as per Keybase convo
